tests/test_ptk_shell.py

```diff
@@ -126,10 +126,17 @@ def test_remove_ansi_osc(raw_prompt, prompt, osc_tokens):
         assert removed == ref
 
 
-def test_ptk_prompt(ptk_shell):
+@pytest.mark.parametrize(
+    "line, exp",
+    [
+        [repr("hello"), None],
+        ["2 * 3", "6"],
+    ],
+)
+def test_ptk_prompt(line, exp, ptk_shell, capsys):
     inp, out, shell = ptk_shell
-    text = "hello"
-    inp.send_text(f"{text}\n")  # note: terminate with '\n'
-    result = shell.singleline()
+    inp.send_text(f"{line}\nexit\n")  # note: terminate with '\n'
+    shell.cmdloop()
+    out, err = capsys.readouterr()
     # todo: check rendered output using https://pyte.readthedocs.io/
-    assert result == text
+    assert out.strip() == exp or line
```

Comment:
Again, what's `line` doing here?

Reply:
when exp is not given it is expected to be same as input. 
like when giving strings

Reply:
So you need to put parentheses around the `or` expression

Reply:
Done!